### ISSUE TITLE
Add support for BSON long

### ISSUE THREAD

**Issue by aequasi** [izatop/mongot#1]
https://mongodb.github.io/node-mongodb-native/api-bson-generated/long.html

**Comment by aequasi** [comment]
You work fast! Thanks :)

**Comment by aequasi** [comment]
Not sure if its from mongot, but i wanted to let you know...

Trying to `collection.findOneAndUpdate({_id}, data, {upsert: true});` with a document that has a long gives me a serializer error right now. Saying that `value.getLowBits` doesn't exist. 

Are you by chance "cloning" the long?

**Comment by izatop** [comment]
@aequasi can you show me the code which reproduce the error and are you sure use latest version of mongot?

**Comment by izatop** [comment]
I've tests with findOneAndUpdate/Replace/etc that passes like this:

```typescript
const newObjectId = new ObjectID();
    const res = await collection.findOneAndUpdate(
        {_id: newObjectId},
        collection.factory({_id: newObjectId, name: 'clean', long: Long.fromNumber(Math.random() * 100000000)}),
        {upsert: true}
    );
```

**Comment by izatop** [comment]
By the way, use Collection.factory to prepare documents before insert or upsert (except update $set op).

**Comment by aequasi** [comment]
Ah, i didnt try the factory thing. Let me try that and get back to you

A. Scherer

**Comment by izatop** [comment]
@aequasi i also test it without `factory`. But you should use `factory` to avoid side effects which may be because documents generate default values for collection properties programmatically.

**Comment by aequasi** [comment]
Factory still didnt work for me.... 

My code is heavily abstracted, but ill try to give relevant information

Note that, `data.id` is a `Long`, and not the same as `model._id`
```
@document
export default class User extends SchemaDocument implements DocumentInterface {
    @prop
    public id: Long;
}

public async upsert(data: any): Promise<T> {
        try {
            console.log("{ID: }", data.id, this.collection.name); // Shows a `Long` as `id`
            const result   = await this.collection.findOneAndUpdate(
                {id: data.id},
                this.collection.factory(data),
                {upsert: true},
            );
            const model: T = result.get();

            return result.get() as T;
        } catch (e) {
            this.kernel.emit("error", e);
        }
    }
```

**Comment by aequasi** [comment]
And the exception:

```
app_1    | [1] 2017-08-30T13:59:32.109Z - error:  TypeError: value.getLowBits is not a function
app_1    | [1]     at serializeLong (/app/node_modules/bson/lib/bson/parser/serializer.js:337:23)
app_1    | [1]     at serializeInto (/app/node_modules/bson/lib/bson/parser/serializer.js:780:17)
app_1    | [1]     at serializeObject (/app/node_modules/bson/lib/bson/parser/serializer.js:308:18)
app_1    | [1]     at serializeInto (/app/node_modules/bson/lib/bson/parser/serializer.js:776:17)
app_1    | [1]     at BSON.serialize (/app/node_modules/bson/lib/bson/bson.js:58:27)
app_1    | [1]     at Query.toBin (/app/node_modules/mongodb-core/lib/connection/commands.js:140:25)
app_1    | [1]     at Pool.write (/app/node_modules/mongodb-core/lib/connection/pool.js:984:23)
app_1    | [1]     at Server.command (/app/node_modules/mongodb-core/lib/topologies/server.js:537:15)
app_1    | [1]     at Server.command (/app/node_modules/mongodb/lib/server.js:383:17)
app_1    | [1]     at executeCommand (/app/node_modules/mongodb/lib/db.js:312:19)
app_1    | [1]     at Db.command (/app/node_modules/mongodb/lib/db.js:336:44)
app_1    | [1]     at findAndModify (/app/node_modules/mongodb/lib/collection.js:2491:13)
app_1    | [1]     at Collection.findAndModify (/app/node_modules/mongodb/lib/collection.js:2423:44)
app_1    | [1]     at findOneAndUpdate (/app/node_modules/mongodb/lib/collection.js:2379:8)
app_1    | [1]     at /app/node_modules/mongodb/lib/collection.js:2363:5
app_1    | [1]     at Promise (<anonymous>)
```

**Comment by aequasi** [comment]
Noticed this:

```
console.log(data.id) // Shows a long
console.log(collection.factory(data).id) // Shows undefined
```

**Comment by izatop** [comment]
@aequasi thank you so much for response. This bug was fixed in `mongot@1.4.14`.

**Comment by aequasi** [comment]
👍
